src/rules/jsxCurlyBracePresenceRule.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isJsxAttribute, isJsxExpression, isStringLiteral, isTextualLiteral } from "tsutils";
+import * as ts from "typescript";
+
+const OPTION_ALWAYS = "always";
+const OPTION_NEVER = "never";
+const CURLY_PRESENCE_VALUES = [OPTION_ALWAYS, OPTION_NEVER];
+const CURLY_PRESENCE_OBJECT = {
+    enum: CURLY_PRESENCE_VALUES,
+    type: "string",
+};
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "jsx-curly-brace-presence",
+        description: "Enforce curly braces or disallow unnecessary curly braces in JSX props",
+        optionsDescription: Lint.Utils.dedent`
+One of the following two options must be provided:
+
+* \`"${OPTION_ALWAYS}"\` requires JSX attributes to have curly braces around string literal values
+* \`"${OPTION_NEVER}"\` requires JSX attributes to NOT have curly braces around string literal values`,
+        options: {
+            type: "array",
+            items: [CURLY_PRESENCE_OBJECT],
+            minLength: 1,
+            maxLength: 1,
+        },
+        optionExamples: [
+            `[true, "${OPTION_ALWAYS}"]`,
+            `[true, "${OPTION_NEVER}"]`,
+        ],
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_CURLY_PRESENT = "JSX attribute must NOT have curly braces around string literal";
+    public static FAILURE_CURLY_MISSING = "JSX attribute must have curly braces around string literal";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const option = Array.isArray(this.ruleArguments) ? this.ruleArguments[0] : undefined;
+
+        return this.applyWithFunction(sourceFile, walk, option);
+    }
+}
+
+function walk(ctx: Lint.WalkContext<string | undefined>): void {
+    return ts.forEachChild(ctx.sourceFile, validateCurlyBraces);
+
+    function validateCurlyBraces(node: ts.Node): void {
+        if (isJsxAttribute(node)) {
+            if (ctx.options === OPTION_ALWAYS) {
```

Comment:
I think this rule should have an opinionated default so that it actually does something when enabled. I would like to default to `{ "props": "never" }`

Reply:
Replaced `else if()` with `else` to implement it. Or is there some way to setup default values for the options?

Reply:
I added a test for the default case, as well.